src/Halogen/VDom/Types.purs

```diff
@@ -72,14 +74,21 @@ runGraft =
     in
       go v
 
-data ElemSpec a = ElemSpec (Maybe Namespace) String a
+data ElemSpec a = ElemSpec (Maybe Namespace) ElemName a
 
 instance functorElemSpec ∷ Functor ElemSpec where
   map f (ElemSpec ns name a) = ElemSpec ns name (f a)
 
-newtype Namespace = Namespace String
+newtype ElemName = ElemName String
```

Comment:
I know you can't run the test app since I haven't committed instructions for doing so 😆 but I don't think it will compile now.